core/src/main/java/org/elasticsearch/search/profile/CollectorResult.java

```diff
@@ -148,9 +148,4 @@ public void writeTo(StreamOutput out) throws IOException {
             child.writeTo(out);
         }
     }
-
-    @Override
-    public Object readFrom(StreamInput in) throws IOException {
-        return new CollectorResult(in);
-    }
```

Comment:
nit: should we move down the corresponding ctor like we do in most of the other cases?